src/Events/Custom_Tables/V1/Updates/Events.php

```diff
@@ -139,16 +140,24 @@ public function rebuild_known_range() {
 		$latest   = $this->get_latest_occurrence();
 
 		if ( $earliest ) {
+			$earliest_id = $earliest->provisional_id;
+			if ( ! $earliest_id ) {
+				$earliest_id = $earliest->post_id;
+			}
```

Comment:
This _could_ be a ternary:
```suggestion
			$earliest_id = ! empty( $earliest->provisional_id ) ? $earliest->provisional_id : $earliest->post_id;
```

Reply:
Just not a null coalescing check because (I'm assuming) it's returning a non-null value (like...false?)

Reply:
It's returning the proper value. This seems to be a bug with null coalescing and magic properties. It returns a positive integer (not `false` or `0` etc) but fails when in a coalescing statement, and seems fine with the traditional ternary check. 

See I checked the return values (it should be a provisional ID): 

![image](https://github.com/the-events-calendar/the-events-calendar/assets/2826045/b61ada40-07bc-4ce7-8ec5-4f913444ae66)